Dockerfile_mbtci_java17

```diff
@@ -1,93 +1,189 @@
+      61FC681DFB92A079F1685E77973F295594EC4689 \
+      8FCCA13FEF1D0C2E91008E09770F7A9A5AE15600 \
+      C4F0DFFF4E8C1A8236409D08E73BC641CC11F4C8 \
+      890C08DB8579162FEE0DF9DB8BEAB4DFCF555EF4 \
+      C82FA3AE1CBEDC6BE46B9360C43CEC45C17AB93C \
+      108F52B48DB57BB0CC439B2997B01419BD92F80A \
+    ; do \
+      gpg --batch --keyserver hkps://keys.openpgp.org --recv-keys "$key" || \
+      gpg --batch --keyserver hkps://keyserver.ubuntu.com --recv-keys "$key" ; \
+    done \
+    # check downloaded archive matches the checksum
+    && curl -fsSLO --compressed "https://nodejs.org/dist/$NODE_VERSION/node-$NODE_VERSION-linux-$ARCH.tar.gz" \
+    && curl -fsSLO --compressed "https://nodejs.org/dist/$NODE_VERSION/SHASUMS256.txt" \
+    && grep " node-$NODE_VERSION-linux-$ARCH.tar.gz\$" SHASUMS256.txt | sha256sum -c - \
+    # check downloaded checksum file signature
+    && curl -fsSLO --compressed "https://nodejs.org/dist/$NODE_VERSION/SHASUMS256.txt.sig" \
+    && gpg --verify SHASUMS256.txt.sig SHASUMS256.txt \
+    && tar -zxv -f "node-$NODE_VERSION-linux-$ARCH.tar.gz" -C /usr/local --strip-components=1 --no-same-owner \
+    && rm "node-$NODE_VERSION-linux-$ARCH.tar.gz" SHASUMS256.txt.sig SHASUMS256.txt \
+    && apt-mark auto '.*' > /dev/null \
+    && find /usr/local -type f -executable -exec ldd '{}' ';' \
+      | awk '/=>/ { print $(NF-1) }' \
+      | sort -u \
+      | xargs -r dpkg-query --search \
+      | cut -d: -f1 \
+      | sort -u \
+      | xargs -r apt-mark manual \
+    && apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false \
+    && ln -s /usr/local/bin/node /usr/local/bin/nodejs \
+    # smoke tests
+    && echo "node ${NODE_VERSION} install smoke tests!" \
+    && node --version \
+    && npm --version
 
-# installing Golang
-	curl -O https://storage.googleapis.com/golang/go${GO_VERSION}.linux-amd64.tar.gz && tar -xvf go${GO_VERSION}.linux-amd64.tar.gz && \
-    mv go /usr/local && \
-    mkdir -p "$GOPATH/src" "$GOPATH/bin" && chmod -R 777 "$GOPATH" && \
-    mkdir -p ${GOPATH}/src ${GOPATH}/bin && \
+# install ui5-cli temporay solution
+RUN	set -ex \
+    && npm install --prefix /usr/local/ -g @ui5/cli \
+    && npm cache clean -g --force \
+    # smoke tests
+    && echo "ui5 install smoke tests!" \
+    && ui5 --version
 
-# update maven home
-	M2_BASE="$(dirname ${M2_HOME})" && \
-    mkdir -p "${M2_BASE}" && \
-    curl --fail --silent --output - "https://apache.osuosl.org/maven/maven-3/${MAVEN_VERSION}/binaries/apache-maven-${MAVEN_VERSION}-bin.tar.gz" \
-    | tar -xzvf - -C "${M2_BASE}" && \
-    ln -s "${M2_HOME}/bin/mvn" /usr/local/bin/mvn && \
-    chmod --recursive a+w "${M2_HOME}"/conf/* && \
+# install Golang
+RUN	set -ex \
+    && apt-get update \
+    && apt-get install -y ca-certificates curl --no-install-recommends \
+    && rm -rf /var/lib/apt/lists/* \
+    && curl -O https://storage.googleapis.com/golang/go${GO_VERSION}.linux-amd64.tar.gz \
```

Comment:
The hardware architecture should be checked, as for other binaries.

Reply:
yes, Done